cmd/tailscale/main.go

```diff
@@ -425,6 +473,31 @@ func (a *App) runBackend() error {
 			}
 		case e := <-backendEvents:
 			switch e := e.(type) {
+			case BugEvent:
+				backendLogID, _ := a.logIDPublicAtomic.Load().(string)
```

Comment:
The API logic should probably all be in it's own self-contained module separate from the backend event handler.  This main.go is already ummm... large.

Now's probably the time to start lumping all of that together.  Don't know exactly how that looks.  Probably in interface and an implementation that defines a method for each localAPI thing you invoke, takes a channel as an argment and parses and writes the response to it directly.

something like

```
...

result := make( chan string) 
err := LocalAPI.GetBugReportID(result); err != nil {
   return err
}

reportId := <-result
a.bugReport <- reportId

...

type LocalAPI interface {
    func (a *localAPI) GetBugReportID(chan result)
}

func (a *LocalAPI) GetBugReportID(chan result) {
     // Do non-web-things things
     result <- reportId
}


```

Maybe even pass the result channel itself around, and/or return it from the API... 

I barely have my head above water when it comes to golang concurrency (let along go in general) so take this all with a grain of salt (except the "put this in it's own module" part).

Reply:
Created a new package for localapi logic, wdyt?